### ISSUE TITLE
Problem of reproducing simulation result (Pusht)

### ISSUE THREAD

**Issue by Mrsirovo** [siddancha/streaming-flow-policy#16]
I tried to reproduce the simulation result on Pusht task, but found in average the success rate is around 73%. Things I have tried:
1. Run the notebook of training and evlauting, resulting in ~92% success rate.
2. Run the same notebook, but seed env.seed as 500, 501, 502, ... as proposed in other scripts, the success rate is ~ 68%.
3. Load the provided checkpoint and evluate as '2', the same.
4. Train sfps and sfpd under streaming-flow-policy, get ~75% and 85%, and the inference time is extremely higher than the notebook.

Thank you in advance for any suggestion.

**Comment by siddancha** [comment]
Hi @Mrsirovo, the training code in the notebook is for illustration and we haven't tuned the hyper-parameters for optimal performance. As you noted, the implementations in `sfps.py` and `sfpd.py` have higher accuracy. The inference time is much higher because it is using `torchdyn.NeuralODE` to perform the flow integration, which is slow. I have two suggestions to speed it up:
- Increase the `atol` and `rtol` parameters of the `NeuralODE` solver to make it converge faster.
- Use the inference code provided in the notebook; this code performs simple but fast Euler integration which is also streamable.

**Comment by fgloris** [comment]
<img width="2830" height="1882" alt="Image" src="https://github.com/user-attachments/assets/32358f18-b595-4445-9680-8e1df81f064a" />
<img width="2545" height="1651" alt="Image" src="https://github.com/user-attachments/assets/9821e3ba-1456-486d-903c-ffab53d0ca17" />
<img width="2830" height="1882" alt="Image" src="https://github.com/user-attachments/assets/d6741d99-0d23-4809-97af-624aba0743ea" />
results of 100 rollouts each, including notebook settings, sfps, sfpd and diffusion. i forked this repo and ran this experiment, but unable to reproduce your results.
